### DISCUSSION TITLE
Creating containers

### DISCUSSION THREAD

**Discussion by anpic** [containers/podman#24585] [Q&A]
Please tell about the different ways to create containers based on the parent linux in which podman is running. It seemed to me that there should be simple ways to create a container using the resources of the base system.

**Comment by rhatdan** [comment]
podman run --help | grep rootfs
      --rootfs                                   The first argument is not an image but the rootfs to the exploded container

Or just look at volumes.  

The question you ask is not very specific.

**Reply by anpic** [reply]
> is not an image but the rootfs to the exploded container

I can't understand it. So is this the path to the image or the path to the file subsystem?


> The question you ask is not very specific.

I want to understand how to quickly create a container without the internet based on the parent linux file system in which podman runs.
Is it possible to quickly create a rootless equivalent of `chroot` with podman?

**Reply by Anachron** [reply]
It's the path to the root of the base-system. 

You need to learn how to build containers:

https://docs.podman.io/en/latest/markdown/podman-build.1.html

**Reply by anpic** [reply]
> It's the path to the root of the base-system.

That's, I can set this path and it will work as the path that is set by chroot?

> You need to learn how to build containers

```
podman-build - Build a container image using a Containerfile
```

Is it possible not to create an image, but to create a temporary virtual container with custom root path?

**Reply by Anachron** [reply]
> That's, I can set this path and it will work as the path that is set by chroot?

Chroot? Podman != chroot. You need to tell podman a path where the filesystem it should use is saved at. You do that with passing ``--rootfs`` to the `podman run` command. 

> Is it possible not to create an image, but to create a temporary virtual container with custom root path?

Yep, in that case, see what @rhatdan suggested: `podman run --help`.

**Reply by anpic** [reply]
> Podman != chroot

similar to chroot

> You need to tell podman a path where the filesystem it should use is saved at

So this path should be writable?

> in that case, see

Do I have to study all the documentation to run the simplest offline container?
There are many ready-made examples for chroot.
